timer/plugin.js

```diff
@@ -38,18 +39,35 @@ export default class CCTimer extends Plugin {
 		Utils.log('[timer] Connected to livesplit');
 		Utils.log('[timer] Loading config..');
 
-		const config = new Config(this.mod);
-		await config.load('settings.json');
+		const configs = [];
 
-		Utils.log('[timer] Loaded config: ', config);
+		//Global Split Settings
+		const mainConfig = new Config(this.mod);
+		await mainConfig.load('settings.json');
+		Utils.log('[timer] Loaded main config: ', mainConfig);
+		configs.push(mainConfig);
 
-		if (config.isIGT) {
+		//Optional Additional Autosplitters (ex. for segments)
+		const fs = require('fs');
+		const configFiles = fs.readdirSync(this.mod.baseDirectory + this.splitsDir);
+
+		for (const file of configFiles) {
+			if(file.endsWith('.json')) {
+				const newConfig = new Config(this.mod);
+				await newConfig.load(this.splitsDir + file);
+				configs.push(newConfig);
+			}
+		}
+
+		console.log(`[timer] Loaded ${configs.length} splits files.`);
+
+		if (mainConfig.isIGT) {
 			Utils.log('[timer] Using original ingame time');
 			this.utils.updateTime(this.connection);
 		} else {
 			Utils.log('[timer] Using custom time filter');
 			const stateManager = new StateManager();
-			stateManager.filterStates(config.filter);
+			stateManager.filterStates(mainConfig.filter);
```

Comment:
As far as I can see this is not updated if another config active (same with `isIGT`)

Reply:
Yeah, the way I have it set up is for `settings.json` to be the only source for time and state settings, I mention it in the readme changes. It looked like it'd be a bit of a hassle to mess with dynamically handling time / state so I decided against it, especially since it's incredibly likely the desired settings would be the same between autosplitters.